### PR TITLE
Make the switch to `docc`

### PR DESCRIPTION
**Pull request by SamWilsn**
#### What changed?

This rips out sphinx and replaces it with docc.

Opening this as a draft because I'm still waiting on a few upstream PRs to merge.

#### Review Suggestions

I very strongly suggest doing this review one commit at a time. Several commits are just fixing formatting/flake8 problems that came up after updating the packages.

#### Cute Animal Picture

cat

### CODE REVIEW

**Review by petertdavies — APPROVED** [review]
I haven't reviewed docc itself, but this all looks good.

File: tests/helpers/load_state_tests.py [hunk 1]
```diff
@@ -103,8 +102,12 @@ def add_block_to_chain(
...
         load.state_transition(chain, block)
     else:
-        with patch(
-            f"ethereum.{load.fork_module}.fork.validate_proof_of_work",
+        fork_module = importlib.import_module(
+            f"ethereum.{load.fork_module}.fork"
+        )
+        with patch.object(
+            fork_module,
+            "validate_proof_of_work",
```
Comment (SamWilsn) [inline tests/helpers/load_state_tests.py:110]: @gurukamath @petertdavies if you could double check that this change maintains the original behaviour, I'd appreciate it!

**Review by petertdavies — APPROVED** [review]
LGTM!

File: src/ethereum_spec_tools/forks.py [hunk 2]
```diff
@@ -61,7 +85,15 @@ def discover(cls: Type[H]) -> List[H]:
...
-            mod = importlib.import_module(pkg.name)
+            if isinstance(pkg.module_finder, importlib.abc.MetaPathFinder):
+                found = pkg.module_finder.find_module(pkg.name, None)
+            else:
+                found = pkg.module_finder.find_module(pkg.name)
+
+            if not found:
+                raise Exception(f"unable to load module {pkg.name}")
+
+            mod = found.load_module(pkg.name)
```
Comment (gurukamath) [inline src/ethereum_spec_tools/forks.py:96]: @SamWilsn After some investigation, I have narrowed it down and it seems to me that these changes (L88 - L96) are causing a memory bloat while loading and running evm tools tests. Not entirely sure why though. Would you have any thoughts?

**Pull request merged**